### PR TITLE
Auth: drop broken support for packet-specific SOA replies from backends

### PR DESCRIPTION
**Pull request by zeha**
### Short description
Remove DNSPacket pointer from getSOA. This was never okay to use (at least with caching enabled).

Split from #4545 per @Habbie's wish.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: modules/luabackend/minimal.cc [hunk 1]
```diff
@@ -176,12 +176,12 @@ bool LUABackend::get(DNSResourceRecord &rr) {
     return !rr.content.empty();
 }
 
-bool LUABackend::getSOA(const DNSName &name, SOAData &soadata, DNSPacket *p) {
+bool LUABackend::getSOA(const string &name, SOAData &soadata) {
```
Comment (Habbie) [inline modules/luabackend/minimal.cc]: `string` ?!

File: modules/luabackend/minimal.cc [hunk 2]
```diff
@@ -176,12 +176,12 @@ bool LUABackend::get(DNSResourceRecord &rr) {
...
+    dnspacket = NULL;
```
Comment (Habbie) [inline modules/luabackend/minimal.cc]: `LUABackend::dnspacket` can be removed completely now.

**Comment by zeha** [comment]
After this force-push battle... my commit does compile now and also sets luapacket=NULL in LUABackend::getSOA(), which is probably important?

**Comment by Habbie** [comment]
`dnspacket=NULL` in getSOA can go away

**Comment by mind04** [comment]
LGTM

**Pull request merged**